index.php

```diff
@@ -50,13 +57,53 @@
 
     $stmt = $this->database->prepare('insert into user (name, password, created_at) values (?, ?, ?)');
     if ($stmt === false) {
-        return $this->renderer->render($response, '/register.php', [
-            'err' => $this->database->lastErrorMsg(),
-        ]);
+        return $this->renderer->render($response, '/register.php', ['errMsg' => $this->database->lastErrorMsg()]);
+    }
+
+    $errs = [];
+
+    if (!isset($body['login'])){
+        $errs[] = "You must register a user name. Please try again.";
+    }
+
+    $stmt2 = $this->database->prepare('select count(name) as number from user where name = ?');  
+    if ($stmt2 === false) {
+        return $this->renderer->render($response, '/register.php', ['errMsg' => $this->database->lastErrorMsg()]);
+    }
+    $stmt2->execute(array($body['login']));
+    $res2 = $stmt2->execute();
+    if ($res2 === false) {
+        return $this->renderer->render($response, '/register.php', ['errMsg' => $this->database->lastErrorMsg()]);
+    }
+
+    $res = $stmt2->fetch();
+
+    if ($res['number'] != 0){
+
+        $errs[] = "User name alreay taken, please select new one.";
+    }
+
+    if ($body['password'] != $body['password_confirmation']){
+        $errs[] = "Password verification failed. Please try again.";
+    } 
+
+    if (strlen($_POST["password"]) < 7) {
+        $errs[] = "Password must exceed 7 characters. Please try again.";
+    } 
+
+    if (count($errs)!=0){
+        return $this->renderer->render($response, '/register.php', ['errs'=> implode('<br/>',$errs)]);
```

Comment:
You should pass an array to the view and let it handle the multiple error messages as it wishes.